src/ngraph/op/strided_slice.hpp

```diff
@@ -0,0 +1,120 @@
+
+#pragma once
+
+#include <memory>
+#include <vector>
+
+#include "ngraph/node.hpp"
+#include "ngraph/op/op.hpp"
+#include "ngraph/op/util/attr_types.hpp"
+
+namespace ngraph
+{
+    namespace op
+    {
+        namespace v1
+        {
+            /// \brief Takes a slice of an input tensor, i.e., the sub-tensor that resides within a
+            ///        bounding box, optionally with stride.
+            class StridedSlice : public Op
+            {
+            public:
+                NGRAPH_API
+                static constexpr NodeTypeInfo type_info{"Slice", 1};
+                const NodeTypeInfo& get_type_info() const override { return type_info; }
+                StridedSlice() = default;
+
+                /// \brief Constructs a dynamic tensor strided slice operation.
+                ///
+                /// \param data             The tensor to be sliced.
+                /// \param begin            1D tensor with begin indexes for input blob slicing.
+                /// \param end              1D tensor with end indexes for input blob slicing.
+                /// \param strides          The slicing strides; for example, strides of `{n,m}`
+                ///                         means to take every nth row and every mth column
+                ///                         of the input matrix.
+                /// \param begin_mask       When begin_mask[i] equal to 1 means that the
+                ///                         corresponding dimension of the begin input is ignored.
+                /// \param end_mask         When end_mask[i] is 1, the corresponding dimension of
+                ///                         the end input is ignored.
+                /// \param new_axis_mask    If new_axis_mask[i] is 1, a length 1 dimension
+                ///                         is inserted on the i-th position.
+                /// \param shrink_axis_mask If shrink_axis_mask[i] is 1, the dimension
+                ///                         on the i-th position is deleted.
+                /// \param ellipsis_mask    It inserts missing dimensions
+                ///                         on a position of a non-zero bit.
+                StridedSlice(const Output<Node>& data,
+                             const Output<Node>& begin,
+                             const Output<Node>& end,
+                             const Output<Node>& strides,
+                             const std::vector<int64_t>& begin_mask,
+                             const std::vector<int64_t>& end_mask,
+                             const std::vector<int64_t>& new_axis_mask = std::vector<int64_t>{},
+                             const std::vector<int64_t>& shrink_axis_mask = std::vector<int64_t>{},
+                             const std::vector<int64_t>& ellipsis_mask = std::vector<int64_t>{});
+
+                /// \brief Constructs a dynamic tensor strided slice operation.
```

Comment:
As I can see, this constructor for the case when strides are optional and have a default value, could you please shortly describe it here?

Reply:
Default value of `strides` is set here: https://github.com/NervanaSystems/ngraph/blob/7f2c9b275e53bf2de230b484db29eb442120e096/src/ngraph/op/strided_slice.cpp#L58 as `Constant` with vector of `1` (length of this vector is input rank -> the same as mask vector size)